rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs

```diff
@@ -501,4 +540,55 @@ mod tests {
         assert!(copied_file_path.exists());
         assert_eq!(fs::read_to_string(copied_file_path).unwrap(), "Hello");
     }
+
+    #[tokio::test]
+    async fn test_wildcard_file_to_nonexistent_directory_destination() {
```

Comment:
The test logic we need to make sure works is cp my_dir/* to existing_dir/, as branches must exist in this method.    Just make sure we have a test covering that.